### PR TITLE
SVERTEXER: Recover ITSonly OB tracks for PCM

### PR DESCRIPTION
**Pull request by f3sch**
Not tested. Don't merge!

### CODE REVIEW

File: Detectors/Vertexing/src/SVertexer.cxx [hunk 1]
```diff
@@ -513,16 +513,19 @@ void SVertexer::buildT2V(const o2::globaltracking::RecoContainer& recoData) // a
...
         if (isITSloaded) {
           auto& itsTrack = recoData.getITSTrack(itsGID);
           nITSclu = itsTrack.getNumberOfClusters();
+          bITSOBonly = itsTrack.getFirstClusterLayer() > 3;
         }
       } else if (itsGID.getSource() == GIndex::ITSAB) {
         if (isITSTPCloaded) {
           auto& itsABTracklet = recoData.getITSABRef(itsGID);
           nITSclu = itsABTracklet.getNClusters();
+          bITSOBonly = itsABTracklet.getFirstEntry() > 3;
```
Comment (shahor02) [inline Detectors/Vertexing/src/SVertexer.cxx:528]: This is anyway guaranteed by the AB track construction

Comment (shahor02) [reply Detectors/Vertexing/src/SVertexer.cxx:528]: The AB tracklet is always in OB and `itsABTracklet.getFirstEntry()` anyway returns not the layer but entry of the 1st cluster in the pool. 
Just use `bITSOBonly = true;`

**Review by shahor02 — CHANGES_REQUESTED** [review]

File: Detectors/Vertexing/src/SVertexer.cxx (hunk 1 above)
Comment (shahor02) [reply Detectors/Vertexing/src/SVertexer.cxx:528]: @ddobrigk has rightfully noted that the AB track is not ITS only by construction, see the continuation of the comment later.

File: Detectors/Vertexing/src/SVertexer.cxx [hunk 2]
```diff
@@ -570,6 +575,7 @@ bool SVertexer::checkV0(const TrackCand& seedP, const TrackCand& seedN, int iP,
   auto& fitterV0 = mFitterV0[ithread];
   // Fast rough cuts on pairs before feeding to DCAFitter, tracks are not in the same Frame or at same X
   bool isTPConly = (seedP.gid.getSource() == GIndex::TPC || seedN.gid.getSource() == GIndex::TPC);
+  bool isITSOBonly = (seedP.bITSOB || seedN.bITSOB);
```
Comment (shahor02) [inline Detectors/Vertexing/src/SVertexer.cxx:578]: Below (L670) you interpret `isITSOBonly` as not only (1) ITS hits being only on the OB but also as (2) not having TPC. 
But to set the bITSOB flag it is enough to fulfill (1) only. Either one should check here additionally for `seedP.gid.getSource() == GIndex::ITS ...` or exclude tracks with TPC in the assignment of the `bITSOB` in the `buildT2V`.

**Review by shahor02 — CHANGES_REQUESTED** [review]

**Comment by f3sch** [comment]
Closed in favor of #12420

**Pull request closed without merging**